packages/ember-data/lib/system/model/model.js

```diff
@@ -690,8 +685,8 @@ var Model = Ember.Object.extend(Ember.Evented, {
   */
   clearRelationships: function() {
     this.eachRelationship(function(name, relationship) {
-      var rel = this._relationships[name];
-      if (rel) {
+      if (this._relationships.has(name)) {
+        var rel = this._relationships.get(name);
```

Comment:
Minor point but if it was never accessed, no point in creating it just in order to clear it